lib/search.rb

```diff
@@ -737,7 +741,7 @@ def posts_query(limit, opts = nil)
 
     is_topic_search = @search_context.present? && @search_context.is_a?(Topic)
 
-    posts = posts.where("topics.visible") unless is_topic_search
+    posts = posts.where("topics.visible") unless is_topic_search || @guardian.is_staff?
```

Comment:
unless with an || makes my brain tingle 

```suggestion
    posts = posts.where("topics.visible") if !is_topic_search && @guardian.is_staff?
```

Reply:
I think it should also be `!guardian.is_staff?` here so I will commit that.